File: study/ts/script/scene/view/equipTrain/EquipLimitCostNode.ts
import HeroLimitCostNode from "../heroTrain/HeroLimitCostNode";
import { EquipTrainHelper } from "./EquipTrainHelper";
import { Path } from "../../../utils/Path";
import { LimitCostConst } from "../../../const/LimitCostConst";
import { G_EffectGfxMgr } from "../../../init";
import UIHelper from "../../../utils/UIHelper";

const { ccclass, property } = cc._decorator;
@ccclass
export default class EquipLimitCostNode extends HeroLimitCostNode {

    changeImageName() {
        var resIds = EquipTrainHelper.getLimitUpCostNameResIds();
        UIHelper.loadTexture(this._imageName, Path.getTextLimit(resIds[this._costKey]));
    }
    _check() {
        this._isShowCount = true;
    }
    playSMoving() {
        var smoving = LimitCostConst.RES_NAME[this._costKey].smoving[1];
        if (this._costKey == LimitCostConst.LIMIT_COST_KEY_2) {
            smoving = 'smoving_tujiehuansanjiao';
        }
        let pos = this.node.getPosition();
        G_EffectGfxMgr.applySingleGfx(this.node, smoving,  ()=> {
            this.node.active = false;
            this.node.setPosition(pos);
        });
    }
    _calPercent(limitLevel, curCount) {
        var info = EquipTrainHelper.getLimitUpCostInfo();
        var size = info['size_' + this._costKey] || 0;
        var percent = Math.floor(curCount / size * 100);
        return [
            Math.min(percent, 100),
            size
        ];
    }
    
    initImageFront() {
    }

    setImageFront(id) {
        UIHelper.loadTexture(this._imageFront, Path.getLimitImg(id));
    }
    setPositionY(y) {
        this._initPos.y = y;
    }

    updateUI(limitLevel, curCount) {
        this.addMask();
        this._updateCommonUI(limitLevel, curCount);
    }
}

File: study/lua/src/app/scene/view/equipTrain/EquipLimitCostNode.lua
local CommonLimitCostNode = require("app.ui.component.CommonLimitCostNode")
local EquipLimitCostNode = class("EquipLimitCostNode", CommonLimitCostNode)
local LimitCostConst = require("app.const.LimitCostConst")
local EquipTrainHelper = require("app.scene.view.equipTrain.EquipTrainHelper")

function EquipLimitCostNode:ctor(target, costKey, callback, limitUpType)
    EquipLimitCostNode.super.ctor(self, target, costKey, callback, limitUpType)
end

function EquipLimitCostNode:changeImageName()
    local resIds = EquipTrainHelper.getLimitUpCostNameResIds()
    self._imageName:loadTexture(Path.getTextLimit(resIds[self._costKey]))
end

function EquipLimitCostNode:_check()
    self._isShowCount = true
end

function EquipLimitCostNode:playSMoving()
    local smoving = LimitCostConst.RES_NAME[self._costKey].smoving[1]
    if self._costKey == LimitCostConst.LIMIT_COST_KEY_2 then
        smoving = "smoving_tujiehuansanjiao"
    end
    G_EffectGfxMgr:applySingleGfx(
        self._target,
        smoving,
        function()
            self._target:setVisible(false)
        end
    )
end

function EquipLimitCostNode:_calPercent(limitLevel, curCount)
    local info = EquipTrainHelper.getLimitUpCostInfo()
    local size = info["size_" .. self._costKey] or 0
    local percent = math.floor(curCount / size * 100)
    return math.min(percent, 100), size
end

function EquipLimitCostNode:setImageFront(id)
    self._imageFront:loadTexture(Path.getLimitImg(id))
end

function EquipLimitCostNode:setPositionY(y)
    self._initPos.y = y
end

return EquipLimitCostNode
